baremetal/metal3data_manager.go

```diff
@@ -784,6 +785,26 @@ func (m *DataManager) releaseAddressFromPool(ctx context.Context, poolName strin
 		addresses[poolName] = true
 		return addresses, false, nil
 	}
+	// Remove finalizer from IPClaim since we no longer need it
+	ipClaim.Finalizers = Filter(ipClaim.Finalizers, infrav1.DataFinalizer)
```

Comment:
we have a `UnsetFinalizer()` method which removes only the finalizers from the m3Data, should we reuse that or have something similar ?

Reply:
I was thinking it is not necessary to have a function for only this. Maybe for ownerRefs, that is a bit longer, but this is just one line anyway.

Reply:
you could rework the same method a bit and make it accept the input so that whenever it is called the corresponding objects could be passed to it. But that is just an improvement of course and can be left from this PR

Reply:
Right, true. I was a bit scared to touch it to be honest since it is a public function that is called from the controller. If it is alright I would like to keep it this way for now.